WordPress/Classes/ViewRelated/Blog/DomainCredit/DomainCreditRedemptionSuccessViewController.swift

```diff
@@ -0,0 +1,69 @@
+import UIKit
+
+@objc protocol DomainCreditRedemptionSuccessViewControllerDelegate {
+    func continueButtonPressed()
+}
+
+/// Displays messaging after user successfully redeems domain credit.
+class DomainCreditRedemptionSuccessViewController: UIViewController {
+    private let domain: String
+
+    private weak var delegate: DomainCreditRedemptionSuccessViewControllerDelegate?
+
+    init(domain: String, delegate: DomainCreditRedemptionSuccessViewControllerDelegate) {
+        self.domain = domain
+        self.delegate = delegate
+        super.init(nibName: nil, bundle: nil)
+    }
+
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        let attributedSubtitleConfiguration: NoResultsViewController.AttributedSubtitleConfiguration = {
+            [weak self] attributedText in
+            guard let `self` = self else {
+                fatalError()
```

Comment:
You can use _self_ with no backticks. We use this approach in many places.
I was thinking, since `attributedSubtitleConfiguration` is an optional argument, can the `AttributedSubtitleConfiguration` return an optional? In this way you don't need to re-bind self but just use the weak reference. What do you think?

Reply:
> You can use self with no backticks. We use this approach in many places.

thanks for pointing it out, now I know - looks like being able to write ``guard let `self` = self else { ... }`` is even a compiler bug according to [Swift evolution](https://github.com/apple/swift-evolution/blob/master/proposals/0079-upgrade-self-from-weak-to-strong.md#relying-on-a-compiler-bug)

> I was thinking, since attributedSubtitleConfiguration is an optional argument, can the AttributedSubtitleConfiguration return an optional? In this way you don't need to re-bind self but just use the weak reference.

just double checking, did you mean `typealias AttributedSubtitleConfiguration = (_ attributedText: NSAttributedString) -> NSAttributedString?`? then going to replace it with `guard let domain = self?.domain else { return nil }`. I also like that this way the configuration block can specify not to alter the attributed subtitle by returning `nil`